### PR TITLE
Use ursa-minor as API proxy

### PR DESCRIPTION
**Pull request by romangraef**
See https://github.com/NotEnoughUpdates/ursa-minor for the server implementation. Does not replace the user profile used by the minion API, since I feel like we should reserve the API proxy only for things that cannot be scraped in game (such as /pv)

Please discuss server design and implementation in [the discord](https://discord.com/channels/516977525906341928/1126315104401641522)

### CODE REVIEW

**Review by NopoTheGamer — DISMISSED** [review]
profile viewer still requries an api key to open it
pv seems to work fine 
There are still general requests trying to use my invalid key but I'm sure you know about those already

**Comment by romangraef** [comment]
> There are still general requests trying to use my invalid key but I'm sure you know about those already

That is correct. The API proxy is only intended for PV, not for things polling passively in the background like skill overlay or minions.

File: src/main/java/io/github/moulberry/notenoughupdates/util/ProfileApiSyncer.java [hunk 1]
```diff
@@ -89,9 +90,15 @@ private void forceResync() {
 		if (Minecraft.getMinecraft().thePlayer == null) return;
 
 		String uuid = Minecraft.getMinecraft().thePlayer.getUniqueID().toString().replace("-", "");
-		NotEnoughUpdates.profileViewer.getOrLoadSkyblockProfiles(uuid, (profile) -> {
-			for (Consumer<SkyblockProfiles> c : finishSyncCallbacks.values()) c.accept(profile);
-			finishSyncCallbacks.clear();
-		});
+		NotEnoughUpdates.INSTANCE.manager.apiUtils
+			.newHypixelApiRequest("/skyblock/profiles")
```
Comment (DeDiamondPro) [inline src/main/java/io/github/moulberry/notenoughupdates/util/ProfileApiSyncer.java:94]: Shouldn't this either use ursa or an anonymous request (not sure which one it should be here)?

Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/util/ProfileApiSyncer.java:94]: Neither is applicable. Ursa minor should be used for the pv (not for passive background polling) and you need an api key to make this request.

**Review by Lulonaut — APPROVED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Pull request merged**